### DISCUSSION TITLE
active_objects vs motion (recording)

### DISCUSSION THREAD

**Discussion by lee-costa** [blakeblackshear/frigate#2906] [Ask A Question] [answered]
Hello, 

While configuring more cameras, I realized that I don't understand the simple difference between `active_objects` and `motion` under the recording settings. 

        #   motion - save all recordings segments with any detected motion
       #   active_objects - save all recording segments with active/moving objects

Does frigate record `motion` regardless of object detected and `active_objects` only when an actual object is detected?

**Comment by NickM-27** [comment, marked as answer]
So with `0.10.0` frigate saves from the recording stream in 10 second segments. These options determine which recording segments are kept for 24/7 recording (but can also affect events). 

Basically, let's say you had your frigate configured so that your doorbell camera would retain the last 2 days of 24/7 recording. 
- With the `all` option all 48 hours of those two days would be kept and viewable. 
- With the `motion` option the only parts of those 48 hours would be segments that frigate detected motion. This is the middle ground option so it doesn't keep everything but keeps segments with motion.
- With `active_objects` it would only keep segments where there was a true positive object that was not considered stationary.

The same options are available with events. Let's consider a scenario where you drive up and park in your driveway, go inside, then come back out 4 hours later.
- With `all` option all segments for the duration of the event would be saved for the event. This event would have 4 hours of footage.
- With `motion` option all segments for the duration of the event with motion would be saved. This means any segment where a car drove by in the street, person walked by, lighting changed, etc. would be saved. 
- With `active_objects` it would only keep segments where the object was active. In this case the only segments that would be saved would be the ones where the car was driving up, you going inside, you coming outside, and the car driving away. Essentially reducing the 4 hours to a minute or two of event footage.

**Reply by lorenzomartini1** [reply]
what is the correct event mode to save events exactly like seen in Frigate events page?

**Reply by NickM-27** [reply]
not sure what you mean, this has to do with what recordings to keep. You should read https://deploy-preview-6262--frigate-docs.netlify.app/configuration/record#what-do-the-different-retain-modes-mean

**Comment by lee-costa** [comment]
Your answer should definitely be part of the documentation! Thank you for the clear explanation.

**Reply by NickM-27** [reply]
@blakeblackshear what do you think? I know there's been a couple users asking about this so perhaps a specific section about retention could help?

**Reply by blakeblackshear** [reply]
That would be great. There is a page for record already.

**Reply by NickM-27** [reply]
Thanks! @junior466 I tried to clean up the wording a little bit for the docs so leave any comments if it doesn't make sense anywhere, PR: https://github.com/blakeblackshear/frigate/pull/2914

**Comment by wjdp** [comment]
@blakeblackshear I'm looking at this page regarding parked cars. Because of camera angles I can't use the zone approach and want to reduce the number of events for parked cars. What I can't tell from the docs is whether `active_objects` only affects retention of events or changes the behaviour of what is considered an event.

I'm gonna try this and see what I get.

(outside of scope of issue but what I feel I'm after is a setting for cars saying they need to move a certain amount rather than staying in same place while some other motion happens in their bounding box)

**Reply by NickM-27** [reply]
It only affects what is recorded not the UI behavior. This is much improved in the next version
